### ISSUE TITLE
List component has incorrect ARIA roles

### ISSUE THREAD

**Issue by mattsherman** [woocommerce/woocommerce-admin#4387]
**Describe the bug**

The `@woocommerce/components` `List` component renders as a `<ul>` with an ARIA role of `menu`. List items are under the `<ul>`'s `<li>`s as `<a>` links with an ARIA role of `menuitem`.

On first glance, I thought this seemed reasonable. But, after thinking about it further and doing some research, I believe that it is incorrect to apply the `menu`/`menuitem` roles for this component.

From [W3C: Using ARIA Menus](https://www.w3.org/WAI/GL/wiki/Using_ARIA_menus):

> A menu widget usually consists of a main menu bar with pop up menus. Items in pop up menus often have sub pop up menus. Menu widgets must provide keyboard support to navigate all the menu items in the menu. Keyboard support for a menu typically include the cursor keys to navigate between the menu items and open and close popupmenus. 

From [Inclusive Components: Menu & Menu Buttons](https://inclusive-components.design/menus-menu-buttons/):

> ARIA menus are not designated for navigation but for application behavior. Imagine the menu system for a desktop application.

Most/all of our use of `<List>` in wc-admin is for navigation. It is not similar to an application menu as found in a typical desktop application.

Examples of `<List>` usage:
- "Set up your store and start selling" task list on dashboard or other screens
- "Store management" quick links card on homepage (in progress as of this issue creation; #4350)

And on top of it using the wrong role, it doesn't even implement the `menu` role fully. Arrow keys are not available for moving between menu items.

**To Reproduce**

1. Go to either the "Set up your store and start selling" task list  or the "Store Management" quick links card on the homepage.
2. Using assistive tool, or by looking at the page source, see that role of `menu`/`menuitem` is used for list.
3. Try to up/down arrow keys to navigate between items. Verify you can't.

**Expected behavior**

Since the usage of `<List>` is for navigation and not application behavior, I expect the `menu`/`menuitem` role to not be used. If anything, these lists should be wrapped in a `<nav>` element.

**Additional context**

- [W3C: Using ARIA menus](https://www.w3.org/WAI/GL/wiki/Using_ARIA_menus)
- [Inclusive Components: Menus & Menu Buttons](https://inclusive-components.design/menus-menu-buttons/)
- [Don't Use ARIA Menu Roles for Site Nav](https://adrianroselli.com/2017/10/dont-use-aria-menu-roles-for-site-nav.html)

**Comment by samueljseay** [comment]
Just noting the `List` component is now deprecated, so I think we're heading in a good direction with this, but we'll need to update our internal usage to use the new ExperimentalList and add roles / markup as needed to support an aria compliant list of links.
